File: P8_parity.py
nibble_to_bits = [0, 1, 1, 0, 1, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0]
def countSetBits(num):
	nibble = 0
	if (0 == num):
		return nibble_to_bits[0]
	nibble = num & 0xf
	return nibble_to_bits[nibble] + countSetBits(num >> 4)
def getParity(num):
	return countSetBits(num) % 2
n = 1
print("Parity of no", n, " = ", ["even", "odd"][getParity(n)])

File: P8_Parity.c
#include<stdio.h>             
#include<string.h>         
void main(){
        char binary_String[100];
        printf("Enter the String");
        scanf("%s",binary_String);
        int onescount = 0;
        
        for (int i = 0; i<strlen(binary_String); i++)
        {
            if(binary_String[i] == '1'){
                onescount++;
            }
        }
        if (onescount % 2 == 0)
        {
            printf("even parity : %d ones are present in the string",onescount );
        }
        else{
            printf("Odd parity : %d ones are present in the string",onescount);
        }
        
}
